src/Views/IframeView.php

```diff
@@ -1,88 +1,289 @@
 	 *
-	 * This will be use to setup body tag classes.
+	 * @since 2.7.0
+	 * @var int
+	 */
+	protected $autoScroll = 0;
+
+	/**
+	 * Iframe minimum height.
 	 *
 	 * @since 2.7.0
-	 * @var array
+	 * @var bool
+	 */
+	protected $minHeight = null;
+
+	/**
+	 * Unique identifier.
+	 *
+	 * @var string|null
+	 */
+	protected $uniqueId = null;
+
+	/**
+	 * Button title.
+	 *
+	 * @var string|null
+	 */
+	protected $buttonTitle = null;
+
+	/**
+	 * Button color.
+	 *
+	 * @var string|null
+	 */
+	protected $buttonColor = null;
+
+	/**
+	 * Form template.
+	 *
+	 * @var Template
+	 */
+	protected $template = null;
+
+	/**
+	 * Form id.
+	 *
+	 * @var int
+	 */
+	protected $formId = 0;
+
+	/**
+	 * IframeView Constructor
+	 *
+	 * @param Template $template
+	 */
+	public function __construct( $template = null ) {
+		$this->uniqueId    = uniqid( 'give-' );
+		$this->buttonTitle = __( 'Click to donate', 'give' );
+	}
+
+	/**
+	 * Set iframe URL.
+	 *
+	 * @param string $url
+	 *
+	 * @return $this
+	 */
+	public function setURL( $url = null ) {
+		$this->url = add_query_arg(
+			array_merge( [ 'giveDonationFormInIframe' => 1 ] ),
+			$url
+		);
+
+		return $this;
+	}
+
+	/**
+	 * Set whether or not show modal.
+	 *
+	 * @param bool $set
+	 *
+	 * @return $this
 	 */
-	protected $bodyClasses = [ 'give-form-templates' ];
+	public function showInModal( $set ) {
+		$this->modal = $set;
+
+		return $this;
+	}
 
 	/**
-	 * Set document page title.
+	 * Button title.
 	 *
 	 * @param string $title
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setTitle( $title ) {
-		$this->title = $title;
+	public function setButtonTitle( $title ) {
+		$this->buttonTitle = $title;
 
 		return $this;
 	}
 
 	/**
-	 * Set document page title.
+	 * Button color.
 	 *
-	 * @param string $body
+	 * @param string $color
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setBody( $body ) {
-		$this->body = $body;
+	public function setButtonColor( $color ) {
+		$this->buttonColor = $color;
 
 		return $this;
 	}
 
 	/**
-	 * Set document page title.
+	 * Form id.
 	 *
-	 * @param array $classes
+	 * @param int $id
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setBodyClasses( $classes ) {
-		$this->bodyClasses = array_merge( $this->bodyClasses, (array) $classes );
+	public function setFormId( $id ) {
+		$this->formId = $id;
 
 		return $this;
 	}
 
+	/**
+	 * Get iframe HTML.
+	 *
+	 * @return string
+	 */
+	private function getIframeHTML() {
+		ob_start();
+		include $this->template->getLoadingView();
+
+		$loader = sprintf(
+			'<div class="iframe-loader">%1$s</div>',
+			ob_get_clean()
+		);
+
+		$iframe = sprintf(
+			'<iframe
+				name="give-embed-form"
+				%1$s
+				%4$s
+				data-autoScroll="%2$s"
+				onload="Give.initializeIframeResize(this)"
+				style="border: 0;visibility: hidden;%3$s"></iframe>%5$s',
+			$this->modal ? "data-src=\"{$this->url}\"" : "src=\"{$this->url}\"",
```

Comment:
`$this->url` needs sanitization

Reply:
Replied here: https://github.com/impress-org/givewp/pull/4659#discussion_r420878384